pkg/scheduler/framework/types.go

```diff
@@ -1049,19 +1049,46 @@ func (n *NodeInfo) update(pod *v1.Pod, sign int64) {
 	n.Generation = nextGeneration()
 }
 
+func getNonMissingContainerRequests(requests v1.ResourceList, podLevelResourcesSet bool) map[v1.ResourceName]resource.Quantity {
+	nonMissingContainerRequests := make(map[v1.ResourceName]resource.Quantity, 2)
+	if !podLevelResourcesSet {
+		return map[v1.ResourceName]resource.Quantity{
+			v1.ResourceCPU:    *resource.NewMilliQuantity(schedutil.DefaultMilliCPURequest, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(schedutil.DefaultMemoryRequest, resource.DecimalSI),
+		}
+	}
+
+	// DefaultMilliCPURequest serves as the fallback value when both
+	// pod-level and container-level CPU requests are unspecified.
+	if _, exists := requests[v1.ResourceCPU]; !exists {
```

Comment:
In this scenario:

1. Pod has pod-level mem request set, no cpu request set.
2. Pod has two containers, one with cpu request set, one not.

Then, we'll ignore cpu in L1063, and led us to a wrong non0Requests, no?

I guess we can get non0Requests  in L1086 first, and then replace them if pod-level cpu/memory set instead of using this `getNonMissingContainerRequests` ?

Reply:
@AxeZhan: The defaulting logic will take care of propagating the requests at container level to pod-level if the request for a resource is not set at pod-level, but is set at container-level. So this won't be a problem. Let me try to add a comment and also an e2e test for the same.